providers/provisioning.rb

```diff
@@ -0,0 +1,47 @@
+require 'chef/mash'
+
+#include GrafanaCookbook::Provisioning
+
+use_inline_resources if defined?(use_inline_resources)
+
+def whyrun_supported?
+  true
+end
+
+action :create do
+  new_resource.datasource[:name] = new_resource.name
+
+  template ::File.join(node['grafana']['datasources']['provisioning_dir'], new_resource.name + '.yml') do
+    cookbook 'grafana'
+    source 'datasource.yml.erb'
+    user node['grafana']['user']
+    group node['grafana']['group']
+    mode 0600
+    variables(
+      'datasource_name' => new_resource.datasource[:name],
+      'datasource_type' => new_resource.datasource[:type],
+      'datasource_url' => new_resource.datasource[:url],
+      'access' => new_resource.datasource[:access],
+      'organization_id' => new_resource.datasource[:organization_id],
+      'json_data' => new_resource.datasource[:json_data].to_json,
+      'version' => new_resource.datasource[:version],
+      'database' => new_resource.datasource[:database],
+      'user' => new_resource.datasource[:user],
+      'password' => new_resource.datasource[:password],
+      'basic_auth_enabled' => new_resource.datasource[:basic_auth_enabled],
+      'basic_auth_user' => new_resource.datasource[:basic_auth_user],
+      'basic_auth_password' => new_resource.datasource[:basic_auth_password],
+      'with_credentials' => new_resource.datasource[:with_credentials],
+      'is_default' => new_resource.datasource[:is_default],
+      'secure_json_data' =>  new_resource.datasource[:secure_json_data].to_json,
+      'editable' => new_resource.datasource[:editable],
+    )
+#    notifies :stop, 'service[]'
+  end
+end
+
+action :delete do
+  file ::File.join(node['grafana']['datasources']['provisioning_dir'], new_resource.name + '.yml') do
+    action :delete
```

Comment:
It won't delete datasource, just stop adding it

Reply:
it does let me show you that with my kicthen tests.